### PR TITLE
doc: Add note about `JsonNode.elements()` -> `values()` rename

### PR DESCRIPTION
**Pull request by JooHyukKim**
Thought maybe we could help users to have a clue?
#4870

### CODE REVIEW

File: src/main/java/com/fasterxml/jackson/databind/JsonNode.java [hunk 1]
```diff
@@ -1071,6 +1071,8 @@ public boolean hasNonNull(int index) {
      * this node is a JSON Array or Object node. In case of Object node,
      * field names (keys) are not included, only values.
      * For other types of nodes, returns empty iterator.
+     * <p>
+     * Note: In Jackson 3, this method will be renamed to {@code values()}.
```
Comment (JooHyukKim) [inline src/main/java/com/fasterxml/jackson/databind/JsonNode.java:1075]: Either this.... or do both of below

1. `@Deprecated` elements() 
2. Add new `values()` method in 2.19 version

Comment (cowtowncoder) [reply src/main/java/com/fasterxml/jackson/databind/JsonNode.java:1075]: I'll merge this, would accept PR for (2) too -- let's wait bit longer until doing (1) (like 2.20).

**Review by cowtowncoder — APPROVED** [review]

File: src/main/java/com/fasterxml/jackson/databind/JsonNode.java (hunk 1 above)
Comment (cowtowncoder) [reply src/main/java/com/fasterxml/jackson/databind/JsonNode.java:1075]: Actually, one note: I will file another issue for changing return type of `values()` to `Collection<JsonNode>`, to align with other accessors (`properties()`, `propertyNames()`) -- easier to use than `Iterator`.

Comment (cowtowncoder) [reply src/main/java/com/fasterxml/jackson/databind/JsonNode.java:1075]: @JooHyukKim Ok, I decided to revisit #4869, see changed description. So part is done for 2.19, rest for 3.0.

Comment (JooHyukKim) [reply src/main/java/com/fasterxml/jackson/databind/JsonNode.java:1075]: Makes sense thanks!

**Pull request merged**